aten/src/ATen/core/ivalue.h

```diff
@@ -50,6 +50,15 @@ struct GenericDict;
 struct Object;
 struct PyObjectHolder;
 struct EnumHolder;
+struct ComplexHolder : c10::intrusive_ptr_target {
```

Comment:
I'd expect a comment here explaining why ComplexHolder is necessary, and in particular why it has to be ref counted (because sharing reused complex numbers isn't buying you much at all, and so if ivalue had supported unique ownership, that would have been better--the point being it doesn't!)

Reply:
Yeah I guess the reference counting is a by-product of `intrusive_ptr` that makes sense for lists, dictionaries and objects but not really for complex types... 😢

Reply:
@ezyang added a comment (#50883)